### ISSUE TITLE
Cannot find the beacon in phone's default bluetooth setting

### ISSUE THREAD

**Issue by ghost** [NanoBeacon/config-files#11]
I am using the IN100 to create a Bluetooth beacon. I want to detect the IN100 in my phone's default Bluetooth settings, but it doesn't seem to appear there. However, it does show up in other applications such as nRF Connect, Bluetooth Terminal, etc. Can you help me understand why this issue is happening?

**Comment by NanoBeacon** [comment]
Hi there,

Are you using an Android or iPhone. What is the version pf the Android OS or iOs?

**Comment by ghost** [comment]
Hi,
I'm using android phone, version 13. I've also tested it on various versions on 11, 12 & 14.

**Comment by StevePoston** [comment]
Hi Shylendran7801,

Remember it's impossible to connect to a IN100 device, which is usually the devices listed when doing normal bluetooth scanning.

Depending on the application use case, it can be done with the Custom Advertising Settings.

Enable the User Defined Data, and enter "0201**02**".  That '**02'** at the end 'Indicates LE General Discoverable Mode'.

See the attached screenshot and give it a try.  It worked for me with Android 14 with the NanoBeacon BLE Scanner app.

Regards,
Steve

![IN100_BT_Scan2](https://github.com/user-attachments/assets/b48de6f2-c90f-4a83-9f4a-74f2cbf3ca39)

**Comment by ghost** [comment]
Could you please provide SDK required to integrate Bluetooth functionality with in100?

**Comment by StevePoston** [comment]
Have a look at IN100 Host Controller Sample repository. [https://github.com/NanoBeacon/IN100-Host-Controller-Sample](url)

**Comment by NanoBeacon** [comment]
Hi Shylendran,

To clarify on Steve's answer, the IN100 is fully Bluetooth 5.3 compliant. It is, however, a tx only device and it has no rx component. This is only relevant to applications that require a two way connection. This would not affect your ability to detect the beacon.

Can you share how you are scanning for the IN100 beacon (as in how are you filtering in the filter settings)? Is the device running in ram when testing? Is this on the development kit or your own board / third party board? You can also try uninstalling the app and reinstalling it.

**Comment by ghost** [comment]
I am using the InPlay Dev Kit and have tested it in both run-in RAM and burn modes. I cannot find the IN100 in normal mobile Bluetooth scanning, both with and without filters (by name or MAC address). However, I can find it using the nRF app. My problem is why it does not appear in mobile Bluetooth scanning. 
In the zip file, I can find nanobeacon_sdk.sln file. Can i use this file as a bluetooth sdk for IN100? or is the sdk different for bluetooth.

**Comment by StevePoston** [comment]
Hi Shylendran7801,
I'm running Android 14.  Doing a normal bluetooth scan, see the attached images.
![Screenshot_20241121_111201_com_android_settings_Settings$BluetoothSettingsActivity-1](https://github.com/user-attachments/assets/a150cabd-0b75-49d3-9b1b-9eddda3174ef)
Tapping the screen where it says "Other Devices", you should see the following:
![Screenshot_20241121_111205_com_android_settings_SubSettings-2](https://github.com/user-attachments/assets/a538ee60-ecbb-40b0-b00d-3569bb17241f)
Then tapping on one of the devices in the list you should get the following screen:
![Screenshot_20241121_111454_com_android_settings_SubSettings-3](https://github.com/user-attachments/assets/d7c905ad-b133-4ac6-8719-240e251cce3a)

I've attached the nanobeacon config file here also for you to try. (Remove the .txt extension in the filename to load with the nanobeacon config tool)
[test_bt_scan .cfg.txt](https://github.com/user-attachments/files/17843342/test_bt_scan.cfg.txt)

When it comes to using the SDK/API source code, the nanobeacon_sdk.sln is a Microsoft Visual Code C++ file, as the example provided is for running on windows.

If you want to run the example on a microcontroller, you need to base/adapt your main code on the nanobeacon_demo.cpp file, and then provide your own code for a microcontroller's usart with the functions that are in the serial_if.h file.  To handle this application demo running on windows, the inplay_uart.c file does this.

So in summary, write your own routines for the usart and include them in your project for what ever development environment you are using.   An extract of the serial_if.h file is as follows, which need to be written in C/C++ for your relevant microcontroller:

> void serial_break(int on);
> int serial_write(uint8_t *buffer, uint32_t buffer_len);
> int serial_read(uint8_t *buffer, uint32_t buffer_len, int tmo);
> int serial_flush(void);
> int serial_open(const char* portname, DWORD baud);
> void serial_close(void);
> 
Remember to include the paths for the files in the bcn directory when you build your own application on a microcontroller.
